feat(server/http): add server

Davincible opened the pull request.

Davincible committed:
feat(server/http): add server

Davincible commented:
We need to add the logger here

Davincible committed:
chore(server/http): merge main (#12)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* feat(codecs/form): add form codec

* fix(codecs/form): rename pkg forms > form

* fix(codecs/form): tests passing

* chore(codecs/form): refactor tests

* fix(codecs/form): take out reflection

* fix(codecs/form): fix linting issues

* chore(codecs/form): merge main (#11)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* chore(codecs/form): fix linting issues & rename file

yeet

Davincible committed:
feat(server/http): migreate logger, codecs, router

Davincible committed:
Merge branch 'main' into server/http

Davincible committed:
chore(server/http): fix linting issues

Davincible committed:
Merge branch 'server/http' of github.com:go-orb/plugins into server/http

Davincible committed:
fix(server/http): don't log on benchmarks

Davincible committed:
fix(server/http): tests & minor fixes

Davincible committed:
fix(server/http): linting issues

Davincible committed:
feat(server/http): add extra tests

Davincible committed:
chore(server/http): refactor tests & log statements

Davincible committed:
fix(server/http): add double start protection

Davincible committed:
fix(server/http): increase test server stop context deadline

Davincible committed:
chore(server/http): add buggy test

Davincible committed:
fix(server/http): add test for content type & log lines

Davincible committed:
fix(server/http): update comment

Davincible committed:
fix(server/http): remove old mdnsregistry from branch

Davincible committed:
chore(server/http): merge main (#19)

* fix(registry/mdns): rename to go-micro repo

* chore(registry/mdns): merge main (#13)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* feat(codecs/form): add form codec

* fix(codecs/form): rename pkg forms > form

* fix(codecs/form): tests passing

* chore(codecs/form): refactor tests

* fix(codecs/form): take out reflection

* fix(codecs/form): fix linting issues

* chore(codecs/form): merge main (#11)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* chore(codecs/form): fix linting issues & rename file

yeet

* fix(codecs/proto): update content types

* fix(codecs/*): Add more codecs, improve them a little.

* fix(registry/mdns): refactor

* chore(registry/mdns): refactor out

* fix(registry/mdns): delete codecs from branch

* fix(registry/mdns): remove log from branch

* merge main (#16)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* feat(codecs/form): add form codec

* fix(codecs/form): rename pkg forms > form

* fix(codecs/form): tests passing

* chore(codecs/form): refactor tests

* fix(codecs/form): take out reflection

* fix(codecs/form): fix linting issues

* chore(codecs/form): merge main (#11)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* chore(codecs/form): fix linting issues & rename file

yeet

* fix(codecs/proto): update content types

* fix(codecs/*): Add more codecs, improve them a little.

* feat(config/*): Add config from the package.

* fix: No more replaces as we have a commited go.work now

* fix(registry/mdns): Integrate with the new config package.

* fix(codecs/*): some lint fixes.

* fix(codecs/form): update content types

* fix(codecs/form): update content types (#15)

* Update Taskfile.yml

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/form/form.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/jsonpb/jsonpb.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set2_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Suggestions from review

* Suggestions from the master of disaster

* Another PR for the mod.

* Another commit for the mod.

* fix linting issues

* fix(registry/mdns): remove wrong mdns registry from branch

Co-authored-by: René Jochum <rene@jochum.dev>

* chore(registry/mdns): update go.work

* fix(registry/mdns): test stuff

* fix(registry/mdns): log

* fix(registry/mdns): go mod/work chaos

* fix(registry/mdns): fix client lint issues

* chore(registry/mdns): more linting issues

* fix(registry/mdns): refactor, linting, options

* fix(registry/mdns): linting issues

* feat(config): Add config from the package.

* feat(config/*): Add config from the package.

* fix: No more replaces as we have a commited go.work now

* fix(registry/mdns): Integrate with the new config package.

* fix(codecs/*): some lint fixes.

* fix(codecs/form): update content types (#15)

* Update Taskfile.yml

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/form/form.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/jsonpb/jsonpb.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set2_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Suggestions from review

* Suggestions from the master of disaster

* Another PR for the mod.

* Another commit for the mod.

* fix linting issues

Co-authored-by: René Jochum <rene@jochum.dev>

* fix(registry/mdns): Remove it as it has its own PR.

* fix(registry/mdns): fix faulty previous commit (ugh')

* fix(registry/mdns): rename providerfunc

* fix: use types.ConfigData

* fix: disable non-generic linters

* fix: rename orb->go-micro

* fix(config): tests

* chore(registry/mdns): config.go cleanup function order

* chore(registry/mdns): refactor tests

Co-authored-by: René Jochum <rene@jochum.dev>

Davincible committed:
fix(server/http): go.mod, go.work & config migration update

Davincible committed:
fix(server/http): test test coverage param

Davincible committed:
feat(server/http): create entrypoint, migrate to core

Davincible committed:
feat(server/http): move tests out, add protoc plugin

Davincible committed:
chore(server/http): fix linting issues

Davincible committed:
feat(server/http): add extra tests for options

Davincible committed:
fix(server/http): go.mod

Davincible committed:
fix(server/http): go.mod & no pkg/errors

Davincible committed:
chore(server/http): merge main branch (#20)

* fix(registry/mdns): rename to go-micro repo

* chore(registry/mdns): merge main (#13)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* feat(codecs/form): add form codec

* fix(codecs/form): rename pkg forms > form

* fix(codecs/form): tests passing

* chore(codecs/form): refactor tests

* fix(codecs/form): take out reflection

* fix(codecs/form): fix linting issues

* chore(codecs/form): merge main (#11)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* chore(codecs/form): fix linting issues & rename file

yeet

* fix(codecs/proto): update content types

* fix(codecs/*): Add more codecs, improve them a little.

* fix(registry/mdns): refactor

* chore(registry/mdns): refactor out

* fix(registry/mdns): delete codecs from branch

* fix(registry/mdns): remove log from branch

* merge main (#16)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* feat(codecs/form): add form codec

* fix(codecs/form): rename pkg forms > form

* fix(codecs/form): tests passing

* chore(codecs/form): refactor tests

* fix(codecs/form): take out reflection

* fix(codecs/form): fix linting issues

* chore(codecs/form): merge main (#11)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* chore(codecs/form): fix linting issues & rename file

yeet

* fix(codecs/proto): update content types

* fix(codecs/*): Add more codecs, improve them a little.

* feat(config/*): Add config from the package.

* fix: No more replaces as we have a commited go.work now

* fix(registry/mdns): Integrate with the new config package.

* fix(codecs/*): some lint fixes.

* fix(codecs/form): update content types

* fix(codecs/form): update content types (#15)

* Update Taskfile.yml

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/form/form.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/jsonpb/jsonpb.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set2_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Suggestions from review

* Suggestions from the master of disaster

* Another PR for the mod.

* Another commit for the mod.

* fix linting issues

* fix(registry/mdns): remove wrong mdns registry from branch

Co-authored-by: René Jochum <rene@jochum.dev>

* chore(registry/mdns): update go.work

* fix(registry/mdns): test stuff

* fix(registry/mdns): log

* fix(registry/mdns): go mod/work chaos

* fix(registry/mdns): fix client lint issues

* chore(registry/mdns): more linting issues

* fix(registry/mdns): refactor, linting, options

* fix(registry/mdns): linting issues

* feat(config): Add config from the package.

* feat(config/*): Add config from the package.

* fix: No more replaces as we have a commited go.work now

* fix(registry/mdns): Integrate with the new config package.

* fix(codecs/*): some lint fixes.

* fix(codecs/form): update content types (#15)

* Update Taskfile.yml

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/form/form.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/jsonpb/jsonpb.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set2_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Suggestions from review

* Suggestions from the master of disaster

* Another PR for the mod.

* Another commit for the mod.

* fix linting issues

Co-authored-by: René Jochum <rene@jochum.dev>

* fix(registry/mdns): Remove it as it has its own PR.

* fix(registry/mdns): fix faulty previous commit (ugh')

* fix(registry/mdns): rename providerfunc

* fix: use types.ConfigData

* fix: disable non-generic linters

* fix: rename orb->go-micro

* fix(config): tests

* chore(registry/mdns): config.go cleanup function order

* chore(registry/mdns): refactor tests

Co-authored-by: René Jochum <rene@jochum.dev>

Davincible committed:
fix(server/http): add source/http back to go.work

Davincible committed:
feat(server/http): add plugin registration

Davincible committed:
fix(server/http): rename config provider

jochumdev commented on server/http/router/router/pluings.go:
```diff
@@ -0,0 +1,6 @@
+package router
```
Typo in filename

jochumdev commented on server/http/router/router/router.go:
```diff
@@ -0,0 +1,78 @@
+
+// The Router interface as found here is a copy of the Chi router interface.
+// It was selected for it's simplicity compliance with the stdlib.
+// Some methods were removded as they were considered unnecessary for go-micro.
```
removed

jochumdev commented on server/http/utils/ip/ip.go:
```diff
@@ -0,0 +1,68 @@
+// Package ip provides validation and parsing utilities.
```
How does this work with ipv6 ? Seems not?

Davincible replied:
Added tests

jochumdev commented on server/http/utils/tls/tls.go:
```diff
@@ -0,0 +1,161 @@
+}
+
+// GenTlSConfig creates a self signed certificate and returns it in a TSL config.
+func GenTlSConfig(addr ...string) (*tls.Config, error) {
```
Shouldn't it be GenTLSConfig?

jochumdev commented on server/http/utils/wg/wg.go:
```diff
@@ -0,0 +1,45 @@
+// Package wg implements a double waitgroup that can be used to keep track
```
This is very "clever" can you explain more what this is for?

Davincible replied:
Not my idea. It's already a v4 util. I really like it as well. I'm not sure if its at place here in the HTTP package, but should be used somewhere.

It allows you to add to a runtime used globally, but only wait for your added routines, and not all other routines, while at the same time giving the safety that the global process knows about your routines.

jochumdev replied:
So if it's not used you can delete it, right?

jochumdev commented on server/http/utils/wg/wg.go:
```diff
@@ -0,0 +1,45 @@
+// Package wg implements a double waitgroup that can be used to keep track
+// of a global waitgroup at the same time. Adding and stubstracting from
+// the waitgroup will impact both the global and local waiggroups, but waiting
```
Typo waitgroups

jochumdev commented on server/tests/proto/generate.go:
```diff
@@ -0,0 +1,8 @@
+package proto
+
+// Download Google proto HTTP annotation libs
+//go:generate wget -q -O google/api/annotations.proto https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/annotations.proto
```
May switch from wget to curl? wget is not every installed while curl mostly is.

Davincible replied:
We could but it's not important, actually only needs to be done once, which it has been, just put it there for documentation

Davincible committed:
chore(server/http): merge main (#21)

* fix(registry/mdns): rename to go-micro repo

* chore(registry/mdns): merge main (#13)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* feat(codecs/form): add form codec

* fix(codecs/form): rename pkg forms > form

* fix(codecs/form): tests passing

* chore(codecs/form): refactor tests

* fix(codecs/form): take out reflection

* fix(codecs/form): fix linting issues

* chore(codecs/form): merge main (#11)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* chore(codecs/form): fix linting issues & rename file

yeet

* fix(codecs/proto): update content types

* fix(codecs/*): Add more codecs, improve them a little.

* fix(registry/mdns): refactor

* chore(registry/mdns): refactor out

* fix(registry/mdns): delete codecs from branch

* fix(registry/mdns): remove log from branch

* merge main (#16)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* feat(codecs/form): add form codec

* fix(codecs/form): rename pkg forms > form

* fix(codecs/form): tests passing

* chore(codecs/form): refactor tests

* fix(codecs/form): take out reflection

* fix(codecs/form): fix linting issues

* chore(codecs/form): merge main (#11)

* feat(log/json): add logger

* Merge pull request #4 from go-orb/log/json
feat(log/json): add logger

* feat(log/text): add logger (#5)

* deps(log/*): x/exp

* feat(codecs): add proto codec (#7)

* feat(codecs/jsonpb): add marshaler (#8)

* feat(codecs/jsonpb): add marshaler

* fix linting issues

* fix linting issues

* chore(codecs/form): fix linting issues & rename file

yeet

* fix(codecs/proto): update content types

* fix(codecs/*): Add more codecs, improve them a little.

* feat(config/*): Add config from the package.

* fix: No more replaces as we have a commited go.work now

* fix(registry/mdns): Integrate with the new config package.

* fix(codecs/*): some lint fixes.

* fix(codecs/form): update content types

* fix(codecs/form): update content types (#15)

* Update Taskfile.yml

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/form/form.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/jsonpb/jsonpb.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set2_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Suggestions from review

* Suggestions from the master of disaster

* Another PR for the mod.

* Another commit for the mod.

* fix linting issues

* fix(registry/mdns): remove wrong mdns registry from branch

Co-authored-by: René Jochum <rene@jochum.dev>

* chore(registry/mdns): update go.work

* fix(registry/mdns): test stuff

* fix(registry/mdns): log

* fix(registry/mdns): go mod/work chaos

* fix(registry/mdns): fix client lint issues

* chore(registry/mdns): more linting issues

* fix(registry/mdns): refactor, linting, options

* fix(registry/mdns): linting issues

* feat(config): Add config from the package.

* feat(config/*): Add config from the package.

* fix: No more replaces as we have a commited go.work now

* fix(registry/mdns): Integrate with the new config package.

* fix(codecs/*): some lint fixes.

* fix(codecs/form): update content types (#15)

* Update Taskfile.yml

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/form/form.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set1_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/json/json.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/jsonpb/jsonpb.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/tests/set2_test.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/proto/proto.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update codecs/yaml/yaml.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/cli/urfave/urfave.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Update config/source/file/file.go

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Apply suggestions from code review

Co-authored-by: David Brouwer <david.brouwer.99@gmail.com>

* Suggestions from review

* Suggestions from the master of disaster

* Another PR for the mod.

* Another commit for the mod.

* fix linting issues

Co-authored-by: René Jochum <rene@jochum.dev>

* fix(registry/mdns): Remove it as it has its own PR.

* fix(registry/mdns): fix faulty previous commit (ugh')

* fix(registry/mdns): rename providerfunc

* fix: use types.ConfigData

* fix: disable non-generic linters

* fix: rename orb->go-micro

* fix(config): tests

* chore(registry/mdns): config.go cleanup function order

* chore(registry/mdns): refactor tests

Co-authored-by: René Jochum <rene@jochum.dev>

Davincible committed:
fix(server/http): code review comments

adfadsf

Davincible committed:
fix(server/http): fix linting issues

Davincible committed:
fix(CI): remove coverpkg flag

jochumdev committed:
chore(server/http): rebase on main

Davincible committed:
fix(server/http): review comments

Davincible committed:
fix(config/source/file): spererate out codec and file not found error (#23)

Davincible committed:
chore(server/http): merge main (#24)

* fix(config/source/file): spererate out codec and file not found error

* fix(config/source/http): fix path construction from url

Davincible committed:
fix(config/source/file): fix file not found error
